python/extensions.bzl

```diff
@@ -44,6 +56,14 @@ python = module_extension(
     tag_classes = {
         "toolchain": tag_class(
             attrs = {
+                "base_url": attr.string(
+                    mandatory = False,
+                    doc = """\
+The URL that is used to override rules_python DEFAULT_RELEASE_BASE_URL. This rule set downloads
+the different Python binaries from this website. Override this URL if you want to have rules_python
+download the hermetic versions of Python from a different website.
+""",
```

Comment:
Should we use the bazel downloader for this instead?

See https://blog.aspect.dev/configuring-bazels-downloader

Reply:
Great question! I asked the folks on #1172 to get their opinion. I think folks are using base_url, so if we are not going to support it people will have to migrate to this as well.

Reply:
See https://github.com/bazelbuild/rules_python/issues/1172#issuecomment-1542320897